api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseFhirDao.java

```diff
@@ -48,29 +54,49 @@
 @SuppressWarnings("UnstableApiUsage")
 public abstract class BaseFhirDao<T extends OpenmrsObject & Auditable> extends BaseDao implements FhirDao<T> {
 	
-	private final TypeToken<T> typeToken;
+	protected final TypeToken<T> typeToken;
+	
+	private final boolean isRetireable;
+	
+	private final boolean isVoidable;
 	
 	@Autowired
 	@Getter(AccessLevel.PROTECTED)
 	@Setter(AccessLevel.PROTECTED)
 	@Qualifier("sessionFactory")
 	private SessionFactory sessionFactory;
```

Comment:
What would be the rationale for making this public? Also, I'm not sure I understand the second comment?

Reply:
> What would be the rationale for making this public? Also, I'm not sure I understand the second comment?

Sorry about that.
So this is how most, if not all of out DaoImpl tests are structured
```java
public class DummyDaoImpl extends BaseFhirDao {

}

public class DummyDaoImplTest extends BaseModuleContextSensitiveTest {

      @Autowired
      @Qualifier("sessionFactory")
      private SessionFactory sessionFactory;

      private DummyDaoImpl dao;
        
      @Before
      public void setup() {
        dao = new DummyDaoImpl();
        dao.setSessionFactory(sessionFactory);
      }
}
```
So if you want to use the same structure for testing from another different module with fhir2 as a dependency, it won't work because of `@Setter(AccessLevel.PROTECTED)`.
 `dao.setSessionFactory(sessionFactory);` will not be available.

Alternatively, you can rely on spring context by auto-wiring DummyDaoImpl instead of using its an instance (Not sure if it can work)

Reply:
Thanks! Yeah, that makes sense.